src/Components/ShowProduct.tsx

```diff
@@ -0,0 +1,265 @@
+  {
+    jour: "Bamboo Watch",
+    piece: "Bamboo Watch",
+    reference: "Bamboo Watch",
+    facture: "Bamboo Watch",
+    type: 2,
+    compte_general: "Bamboo Watch",
+    compte_tiers: "Bamboo Watch",
+    libelle_ecriture: "Bamboo Watch",
+    date_echeance: "Bamboo Watch",
+    position_journal: "Bamboo Watch",
+    debit: "Bamboo Watch",
+    credit: "Bamboo Watch",
+  },
+  {
+    jour: "Bamboo Watch",
+    piece: "Bamboo Watch",
+    reference: "Bamboo Watch",
+    facture: "Bamboo Watch",
+    type: 3,
+    compte_general: "Bamboo Watch",
+    compte_tiers: "Bamboo Watch",
+    libelle_ecriture: "Bamboo Watch",
+    date_echeance: "Bamboo Watch",
+    position_journal: "Bamboo Watch",
+    debit: "Bamboo Watch",
+    credit: "Bamboo Watch",
+  },
+  {
+    jour: "Bamboo Watch",
+    piece: "Bamboo Watch",
+    reference: "Bamboo Watch",
+    facture: "Bamboo Watch",
+    type: 4,
+    compte_general: "Bamboo Watch",
+    compte_tiers: "Bamboo Watch",
+    libelle_ecriture: "Bamboo Watch",
+    date_echeance: "Bamboo Watch",
+    position_journal: "Bamboo Watch",
+    debit: "Bamboo Watch",
+    credit: "Bamboo Watch",
+  },
+  {
+    jour: "Bamboo Watch",
+    piece: "Bamboo Watch",
+    reference: "Bamboo Watch",
+    facture: "Bamboo Watch",
+    type: 5,
+    compte_general: "Bamboo Watch",
+    compte_tiers: "Bamboo Watch",
+    libelle_ecriture: "Bamboo Watch",
+    date_echeance: "Bamboo Watch",
+    position_journal: "Bamboo Watch",
+    debit: "Bamboo Watch",
+    credit: "Bamboo Watch",
+  },
+  {
+    jour: "Bamboo Watch",
+    piece: "Bamboo Watch",
+    reference: "Bamboo Watch",
+    facture: "Bamboo Watch",
+    type: 6,
+    compte_general: "Bamboo Watch",
+    compte_tiers: "Bamboo Watch",
+    libelle_ecriture: "Bamboo Watch",
+    date_echeance: "Bamboo Watch",
+    position_journal: "Bamboo Watch",
+    debit: "Bamboo Watch",
+    credit: "Bamboo Watch",
+  },
+];
+
+const items = [
+  {
+    label: "Voir les registres associés",
+  },
+  {
+    label: "Visualiser la piece",
+  },
+  {
+    label: "Visionner les documents rattachés",
+  },
+];
+
+export default function ShowProduct() {
+  const navigate = useNavigate();
+  const params = useParams();
+  const [product, setProduct] = useState<Product>({});
+  const [selectedType, setSelectedType] = useState<Type | null>(null);
+  const [products, setProducts] = useState<Product[]>(produit);
+  const [selectedProduct, setSelectedProduct] = useState<Product>();
+  const [ShowProduct, setShowProduct] = useState<Product[]>();
+
+  useEffect(() => {
+    if (params) {
+      let _product = products.filter((val) => val.type === params.type);
+      setShowProduct(_product);
+    }
+  }, [products]);
+
+  const [filters, setFilters] = useState<DataTableFilterMeta>({
+    type: { value: null, matchMode: FilterMatchMode.CONTAINS },
+  });
+  const onGlobalFilterChange = (e: DropdownChangeEvent) => {
+    const { code } = e.target.value;
+    let _filters = { ...filters };
```

Comment:
any reason for declare that with let instead of const ..

Reply:
allright